### DISCUSSION TITLE
Possible to suppress PSRule output in terminal completely and output as a JSON-file

### DISCUSSION THREAD

**Discussion by lassehastrup** [microsoft/PSRule#1863] [Q&A] [answered]
Hi,

I have a scenario where I want to use Pester to test some functionality in PSRule, and therefore I need the output as JSON. 
That's not an issue, and works as expected. However, I would like to NOT receive any form of output from the terminal/pipeline/host when running 'Assert-PSRule' in PowerShell.
I've attempted with most of the options available from the optionFile, but I can't seem to suppress everything and simply just get the output written to a JSON file.

Using variables, Out-Null doesn't work either. My only solution is to start the job in a separate thread using "Start-Job".

Any hints?

**Comment by BernieWhite** [comment, marked as answer]
Hi @lassehastrup, Thanks for the question. For integrating with PowerShell tools try [Invoke-PSRule](https://microsoft.github.io/PSRule/v2/commands/PSRule/en-US/Invoke-PSRule/).

`Assert-PSRule` is intended to provide a CI/CD friendly output to pipelines as text.

`Invoke-PSRule` will output native PowerShell objects that can be tested with Pester. It accepts almost identical parameters and supports `ps-rule.yaml` same as `Assert-PSRule`.

PSRule for Azure has a number of examples of this. See: https://github.com/Azure/PSRule.Rules.Azure/blob/main/tests/PSRule.Rules.Azure.Tests/Azure.ACR.Tests.ps1

---

Also note there is a few deprecations for the future v3 release to be aware of: https://microsoft.github.io/PSRule/v3/deprecations/

---

I hope that answers the question.

**Reply by lassehastrup** [reply]
Exactly what I needed, @BernieWhite . Thank you very much.
